packages/server/modules/core/dbSchema.ts

```diff
@@ -0,0 +1,168 @@
+import knex from '@/db/knex'
+import { Knex } from 'knex'
+import { reduce } from 'lodash'
+
+/**
+ * TODO:
+ * ServerInvites:
+ *  - Get rid of the 'used' field, it's not used anymore
+ */
+
+type SchemaConfig<T extends string, C extends string> = InnerSchemaConfig<T, C> & {
+  /**
+   * Return schema helper with custom configuration options
+   */
+  with: (params?: SchemaConfigParams) => InnerSchemaConfig<T, C>
+}
+
+type InnerSchemaConfig<T extends string, C extends string> = {
+  /**
+   * Table name
+   */
+  name: T
+  /**
+   * Get `knex(tableName)` QueryBuilder instance
+   */
+  knex: () => Knex.QueryBuilder
+  /**
+   * Get names of table columns. The names can be prefixed with the table name or not, depending
+   * on whether `withoutTablePrefix` was set when accessing the helper.
+   */
+  col: {
+    [colName in C]: string
+  }
+}
+
+type SchemaConfigParams = {
+  /**
+   * Configure `col` properties to not have the table name prefixed. For the most part you want the prefix,
+   * cause this helps in queries with JOINS (when multiple tables have a col with the same name), but you don't
+   * want the prefix when triggering UPDATE queries, because the `SET <name> = <value>` syntax doesn't support
+   * column names with table prefixes.
+   */
+  withoutTablePrefix?: boolean
+}
+
+function buildTableHelper<T extends string, C extends string>(
+  tableName: T,
+  columns: C[]
+): SchemaConfig<T, C> {
+  function buildInnerSchemaConfig(
+    params: SchemaConfigParams = {}
+  ): InnerSchemaConfig<T, C> {
+    return {
+      name: tableName,
+      knex: () => knex(tableName),
+      col: reduce(
+        columns,
+        (prev, curr) => {
+          prev[curr] = params.withoutTablePrefix ? curr : `${tableName}.${curr}`
+          return prev
+        },
+        {} as Record<C, string>
+      )
+    }
+  }
+
+  return {
+    ...buildInnerSchemaConfig(),
+    with: buildInnerSchemaConfig
+  }
+}
+
+/*
+ * TABLE RECORD TYPES
+ */
+
+export type ServerInviteRecord = {
+  id: string
+  target: string
+  inviterId: string
+  createdAt?: Date
+  used?: boolean
+  message?: string
+  resourceTarget?: string
+  resourceId?: string
+  role?: string
+  token: string
+}
+
+/*
+ * TABLE HELPERS
```

Comment:
Basically just re-did these with TypeScript to get vastly improved DX with a lot less code - to define a new helper you just have to invoke `buildTableHelper` and pass in the table and column names